src/NumField/NfAbs/NonSimple.jl

```diff
@@ -917,7 +917,11 @@ function number_field(f::Vector{QQPolyRingElem}, s::String="_\$"; cached::Bool =
   return number_field(f, lS, cached = cached, check = check)
 end
 
-function number_field(f::Vector{QQPolyRingElem}, s::Vector{String}; cached::Bool = false, check::Bool = true)
+function number_field(f::Vector{QQPolyRingElem}, s::VarName; cached::Bool = false, check::Bool = true)
+  return number_field(f, String(s); cached = cached, check = check)
```

Comment:
```suggestion
  return number_field(f, Symbol(s); cached = cached, check = check)
```
I think this is the way you do it most other times

Reply:
No, this code deliberately uses the `String` case as base case.

Reply:
Specifically, the base case starts in line 910 and is hidden here, so that's what it looks like:
```julia
function number_field(f::Vector{QQPolyRingElem}, s::String="_\$"; cached::Bool = false, check::Bool = true)
  n = length(f)
  if occursin('#', s)
    lS = Symbol[Symbol(replace(s, "#"=>"$i")) for i=1:n]
  else
    lS = Symbol[Symbol("$s$i") for i=1:n]
  end
  return number_field(f, lS, cached = cached, check = check)
end
```
So it performs string manipulation on `s`.

Reply:
Ah ok, thanks